### PR TITLE
ch-image: update pull behavior

### PR DESCRIPTION
**Pull request by j-ogas**
Addresses #1305.

This enables `ch-image` to catch and pull changes from images in external repositories.

### CODE REVIEW

File: lib/pull.py
Comment (reidpr) [inline lib/pull.py:146]: These two comments should explain why we always download it.

File: lib/pull.py
Comment (reidpr) [inline lib/pull.py:115]: This should use the hash in the filename rather than re-computing it. In fact, does this even need to be changed? By the time we call it, won't the config path be up to date with the new hash from the manifest?

**Review by reidpr — CHANGES_REQUESTED** [review]
Couple suggestions in-line.

Is there a way to add a test that's not too onerous? Perhaps in `build/50_localregistry.bats`?

**Comment by adrecord** [comment]
Any updates on this one?  This causes real problems when the pulled image can't be trusted as being correct.  I currently recommend people always use `--no-cache` to be sure.

**Comment by reidpr** [comment]
Yes, I want to wait until the build cache PR lands because that will upend that part of the code. But this is one of the first things to merge after that. (Thanks for waiting on my reply; I've been on vacation.)

**Comment by j-ogas** [comment]
OK, this is now behaving as intended. `ch-image pull` now always retrieves the image manifests (fat and standard) and will download any config and layers referenced in the manifest should they not already exist in storage.

I do wonder now if this behavior makes `--dlcache` obsolete and thus unnecessary. It was previously used to manually force an update by re-downloading everything. Now images in storage will always be up-to-date. Is there any value in forcing a re-download of everything? FWIW, removing `--dlcache` does make things much simpler, e.g., we could simply replace `--bucache` with `--cache`?

File: doc/ch-image_desc.rst
Comment (j-ogas) [inline doc/ch-image_desc.rst:89]:
```suggestion
    Do not reuse cached operations; execute all non :code:`FROM` instructions and add them to the cache.  
```

**Review by reidpr — APPROVED** [review]
LGTM. Nice work! If tests pass and my tidy looks OK to you, then I'll merge.

File: test/build/55_cache.bats
Comment (j-ogas) [inline test/build/55_cache.bats:465]:
```suggestion
#FIXME: for issue: 1359, add test here where they revert the image in the remote registry to a previous state; our next pull will hit, and so too should any subsequent previously cached instructions based on the FROM SID.```

**Review by j-ogas — COMMENTED** [review]
A couple minor suggestions. Once resolved I'm happy with merging.

**Pull request merged**
